### PR TITLE
feat: Make filters and dividers display horizontally in horizontal native filters filter bar

### PR DESCRIPTION
**Pull request by codyml**
### SUMMARY

This PR adds filters and dividers to the horizontal filter bar, based off of @kgabryje's #22042 with some additional styling work on top.  Dividers and filter wrappers (labels + spacing) should look correct, though the filter input controls themselves are not yet styled in this PR.  Known issues that will be covered in future PRs:
- Filter control inputs aren't updated for horizontal layout
- "More filters" dropdown has incorrect icon, label, badge numbers and doesn't open automatically when related filter links are clicked
- Overflow container may have some resizing quirks
- Hover labels for dividers in horizontal overflow aren't showing up
- Filter cards aren't positioned quite right

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags: `HORIZONTAL_FILTER_BAR`
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControl.tsx [hunk 1]
```diff
@@ -129,27 +231,60 @@ const FilterControl: React.FC<FilterProps> = ({
...
+    return FilterCardPlacement.Right;
+  }, [orientation, overflow]);
 
   return (
-    <StyledFilterControlContainer layout="vertical">
-      <FilterCard filter={filter} isVisible={!isFilterActive && !isScrolling}>
+    <FilterControlContainer
+      layout={
+        orientation === FilterBarOrientation.HORIZONTAL && !overflow
+          ? 'horizontal'
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControl.tsx:279]: nit: should we use an enum for `horizontal` and `vertical` layout? Maybe we could even simply use `FilterBarOrientation` for that?

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterDivider.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  return (
+    <div
+      css={css`
+        border-top: 1px solid ${theme.colors.grayscale.light2};
+        padding-top: ${theme.gridUnit * 4}px;
+        margin-bottom: ${theme.gridUnit * 4}px;
+      `}
+    >
+      <Tooltip overlay={titleIsTruncated ? <strong>{title}</strong> : null}>
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterDivider.tsx:111]: Should the tooltip content be bold? Can't see that requirement in the designs

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterValue.tsx [hunk 3]
```diff
@@ -270,7 +277,7 @@ const FilterValue: React.FC<FilterProps> = ({
           height={HEIGHT}
           width="100%"
           showOverflow={showOverflow}
-          formData={formData}
+          formData={formDataWithDisplayParams}
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterValue.tsx:280]: hmmm I'm not sure if orientation and overflow belongs to `formData`... I'd rather add a new optional prop to `SuperChart` like `displaySetting: Record<string, any>` and put those values there.

**Comment by kgabryje** [comment]
A few more comments:

1. If divider has only title, should the tootlip for title be displayed when hovering over title instead of that icon? 
image

2. Truncated dividers in the dropdown don't show the tooltip on hover 
image

3. If there are a lot of filters in the dropdown, I think it should have some max height and be scrollable + it should close itself when user starts scrolling the dashboard to avoid the effect of floating box

https://user-images.githubusercontent.com/15073128/203841499-7a1a2018-6c1b-4a07-a15d-cdeb0ee614a0.mov

**Comment by kgabryje** [comment]
However, those are all minor problems and this PR looks great! I'm happy to merge it now and address the comments in separate PRs

**Review by kgabryje — APPROVED** [review]
The comments are non blocking since this is locked behind a feature flag. LGTM

**Review by geido — APPROVED** [review]

**Pull request merged**